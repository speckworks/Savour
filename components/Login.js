import React from "react";
import { formStyles } from "../styles/global";
import { View, Text, ScrollView, TextInput } from "react-native";
import { AUTH_TOKEN } from "../constants";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";

const SIGNUP_MUTATION = gql`
  mutation SignupMutation($email: String!, $password: String!, $name: String!) {
    signup(email: $email, password: $password, name: $name) {
      token
    }
  }
`;

const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`;
class Login extends React.Component {
  state = {
    login: true,
    email: "",
    password: "",
    name: "",
  };

  _confirm = async (data) => {
    const { token } = this.state.login ? data.login : data.signup;
    this._saveUserData(token);
  };
  _saveUserData = (token) => {
    localStorage.setItem(AUTH_TOKEN, token);
  };

  render() {
    const { login, email, name, password } = this.state;
    return (
      <ScrollView>
        <Text>{login ? "Login" : "Sign Up"}</Text>
        <View>
          {!login && (
            <TextInput
              style={formStyles.input}
              value={name}
              onChange={(e) => this.setState({ name: e.target.value })}
              type="text"
              placeholder="Your name"
            />
          )}
          <TextInput
            style={formStyles.input}
            value={email}
            onChange={(e) => this.setState({ email: e.target.value })}
            type="text"
            placeholder="Your email address"
          />
          <TextInput
            style={formStyles.input}
            value={password}
            onChange={(e) => this.setState({ password: e.target.value })}
            type="password"
            placeholder="Choose a safe password"
          />
        </View>
        <View>
          <Mutation
            mutation={login ? LOGIN_MUTATION : SIGNUP_MUTATION}
            variables={{ email, password, name }}
            onCompleted={(data) => this._confirm(data)}
          >
            {(mutation) => (
              <Text onClick={mutation}>
                {login ? "login" : "create account"}
              </Text>
            )}
          </Mutation>
          <Text
            style={formStyles.links}
            onClick={() => this.setState({ login: !login })}
          >
            {login ? "need to create an account?" : "already have an account?"}
          </Text>
        </View>
      </ScrollView>
    );
  }


}
export default Login;
