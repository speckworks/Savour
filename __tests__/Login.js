import React from "react";
import Login from "../components/Login";
import renderer from "react-test-renderer";
import gql from "graphql-tag";
import { useQuery } from "@apollo/react-hooks";
import { MockedProvider } from '@apollo/react-testing';

test('renders correctly', () => {
  const tree = renderer.create(
    <MockedProvider mocks={[]} addTypename={false}>
      <Login/>
      </MockedProvider>).toJSON();
  expect(tree).toMatchSnapshot();
});
const component = renderer.create(
  <MockedProvider mocks={[]} addTypename={false}>
    <Login />
  </MockedProvider>,
);

// describe("<Login />", () => {
//   context("when new users opens app", () => {
//     it("displays the new user signup", ()=> component.expect());
//   });
//   context("when prev users opens app", () => {
//     it("displays the existing user login");
//   });
//   context("when new user submits sign up", () => {
//     it("returns JSON token and updates login success");
//   });
// });
