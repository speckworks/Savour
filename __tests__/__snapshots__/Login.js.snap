// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component testing for <login> <Login> renders 1`] = `
<RCTScrollView>
  <View>
    <View>
      <Text>
         Login component 
      </Text>
      <Text>
        Email:
        <TextInput
          allowFontScaling={true}
          id="email"
          name="email"
          onChange={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "flex": 1,
              "padding": 20,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </Text>
      <Text>
        Password:
        <TextInput
          allowFontScaling={true}
          id="password"
          name="password"
          onChange={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "flex": 1,
              "padding": 20,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
